delta_go/state.py

```diff
@@ -0,0 +1,53 @@
+import copy
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Tuple
+
+import numpy as np
+
+from liberty_tracker import LibertyTracker
+from utils import BLACK, EMPTY_BOARD, PlayerMove
+
+
+@dataclass
+class State:
+    """Describes the State of a game of Go.
+
+    Args:
+        board: [BOARD_SIZE, BOARD_SIZE] np array of ints
+        recent_moves: a tuple of PlayerMoves, such that recent[-1] is the last move.
+        to_play: BLACK or WHITE
+        player_color: Keeps track of white color (BLACK or WHITE) you are playing as
+
+        ko: a tuple (x, y) if the last move that was a ko, or None if no ko
+        board_deltas: a np.array of shape (n, go.N, go.N) representing changes
+            made to the board at each move (played move and captures).
+            Should satisfy next_pos.board - next_pos.board_deltas[0] == pos.board
+        lib_tracker: a LibertyTracker object. Used for caching available liberties for speedup.
+                    Gives a speedup of 5x!
+    """
+
+    board: np.ndarray = EMPTY_BOARD
+    recent_moves: Tuple[PlayerMove, ...] = tuple()
+    to_play: int = BLACK
+    player_color: int = BLACK
```

Comment:
As discussed, I think this should be removed and simply made part of the `GoEnv` object - it doesn't need to be tracked when running episodes functionally (i.e. in MCTS) where there is no "player"